src/lib/communication/DY.Lib.Communication.Core.Lemmas.fst

```diff
@@ -52,33 +53,32 @@ val initialize_communication_proof:
     has_pki_invariant
   )
   (ensures (
-    let (_, tr_out) = initialize_communication sender receiver tr in
+    let (_, tr_out) = initialize_communication_core a sender receiver tr in
     trace_invariant tr_out
   ))
   [SMTPat (trace_invariant #invs tr);
-   SMTPat (initialize_communication sender receiver tr);
+   SMTPat (initialize_communication_core a sender receiver tr);
   ]
-let initialize_communication_proof tr sender receiver =
-  reveal_opaque (`%initialize_communication) (initialize_communication sender receiver)
+let initialize_communication_core_proof tr a sender receiver =
+  reveal_opaque (`%initialize_communication_core) (initialize_communication_core a sender receiver)
 #pop-options
 
 (**** Confidential Send and Receive Lemmas ****)
 
 val encrypt_message_proof:
   {|cinvs:crypto_invariants|} ->
-  #a:Type -> {| parseable_serializeable bytes a |} ->
+  #a:Type -> {|comm_layer_core_config a|} ->
   tr:trace ->
   sender:principal -> receiver:principal ->
   pk_receiver:bytes -> nonce:bytes -> payload:a ->
   Lemma
   (requires (
-    let payload_bytes = serialize #bytes a payload in
-    has_communication_layer_crypto_predicates /\
+    has_communication_layer_core_crypto_predicates a /\
     is_secret (long_term_key_label sender) tr nonce /\
     nonce `has_usage tr` PkeNonce /\
-    is_public_key_for tr pk_receiver (LongTermPkeKey comm_layer_pkenc_tag) receiver /\
-    is_knowable_by (join (principal_label sender) (principal_label receiver)) tr payload_bytes /\
-    event_triggered tr sender (CommConfSendMsg sender receiver payload_bytes)
+    is_public_key_for tr pk_receiver (LongTermPkeKey (comm_layer_pkenc_tag a)) receiver /\
+    is_well_formed a (is_knowable_by (join (principal_label sender) (principal_label receiver)) tr) payload /\
```

Comment:
While we're at this, should this `join (principal_label sender) (principal_label receiver)` be replaced with `comm_label sender receiver`?